Regarding obsolete versions

In the earlier days of FreeTDS, Microsoft did not release official specs for the TDS protocol. When MSSQL 2000 (product 8.0) was released, there was semi-official indications from the Microsoft community that the TDS protocol would be version 8.0. So the FreeTDS developers adopted that version for FreeTDS. Years later, when Microsoft started releasing official specs of the protocol, it became obvious that the TDS versions that FreeTDS had labeled 8.0 and 9.0 were actually versions 7.1 and 7.2 respectively.

http://www.freetds.org/userguide/choosingtdsprotocol.htm#AEN910

diff --git a/src/cmd_connect.c b/src/cmd_connect.c
index e1d35cb..2956050 100644
--- a/src/cmd_connect.c
+++ b/src/cmd_connect.c
@@ -860,8 +860,8 @@ int cmd_connect( argc, argv )
         /* Then we use freetds which uses enum instead of defines */
         else if (strcmp(tds_version, "7.0") == 0)
             version = CS_TDS_70;
-        else if (strcmp(tds_version, "8.0") == 0)
-            version = CS_TDS_80;
+        else if (strcmp(tds_version, "7.1") == 0)
+            version = CS_TDS_71;
 #endif
         else version = CS_TDS_50; /* default version */
 
@@ -1258,8 +1258,8 @@ int cmd_connect( argc, argv )
                 case CS_TDS_70:
                     env_set( g_env, "tds_version", "7.0" );
                     break;
-                case CS_TDS_80:
-                    env_set( g_env, "tds_version", "8.0" );
+                case CS_TDS_71:
+                    env_set( g_env, "tds_version", "7.1" );
                     break;
 #endif
                 default:
